2025-10-29 19:13:03,830 INFO: Blackjack Simulator startup [in /home/user/jostmva/frontend/blackjack_simulator/app.py:165]
2025-10-29 19:13:25,583 ERROR: Exception on /simulation/1/run_action [POST] [in /home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py:838]
Traceback (most recent call last):
  File "/home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/jostmva/frontend/blackjack_simulator/routes.py", line 98, in run_simulation_action
    "rules": casino.rules
             ^^^^^^^^^^^^
AttributeError: 'Casino' object has no attribute 'rules'
2025-10-29 19:15:10,264 INFO: Blackjack Simulator startup [in /home/user/jostmva/frontend/blackjack_simulator/app.py:165]
2025-10-29 19:15:14,344 INFO: Blackjack Simulator startup [in /home/user/jostmva/frontend/blackjack_simulator/app.py:165]
2025-10-29 19:15:33,859 ERROR: Exception on /simulation/1/run_action [POST] [in /home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py:838]
Traceback (most recent call last):
  File "/home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/jostmva/frontend/blackjack_simulator/routes.py", line 98, in run_simulation_action
    "rules": casino.rules
             ^^^^^^^^^^^^
AttributeError: 'Casino' object has no attribute 'rules'
2025-10-29 19:18:55,279 INFO: Blackjack Simulator startup [in /home/user/jostmva/frontend/blackjack_simulator/app.py:165]
2025-10-29 19:20:29,370 INFO: Blackjack Simulator startup [in /home/user/jostmva/frontend/blackjack_simulator/app.py:165]
2025-10-29 19:20:33,527 INFO: Blackjack Simulator startup [in /home/user/jostmva/frontend/blackjack_simulator/app.py:165]
2025-10-29 19:21:48,630 INFO: Running simulation 1 [in /home/user/jostmva/frontend/blackjack_simulator/routes.py:72]
2025-10-29 19:21:48,635 ERROR: Exception on /simulation/1/run_action [POST] [in /home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py:838]
Traceback (most recent call last):
  File "/home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/jostmva/frontend/blackjack_simulator/routes.py", line 76, in run_simulation_action
    sim.casino_.id = request.form.get('casino_id')
    ^^^^^^^^^^^
AttributeError: 'Simulation' object has no attribute 'casino_'. Did you mean: 'casino'?
2025-10-29 19:24:33,981 INFO: Blackjack Simulator startup [in /home/user/jostmva/frontend/blackjack_simulator/app.py:165]
2025-10-29 19:24:38,627 INFO: Blackjack Simulator startup [in /home/user/jostmva/frontend/blackjack_simulator/app.py:165]
2025-10-29 19:24:58,280 INFO: Running simulation 1 [in /home/user/jostmva/frontend/blackjack_simulator/routes.py:72]
2025-10-29 19:24:58,288 ERROR: Exception on /simulation/1/run_action [POST] [in /home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py:838]
Traceback (most recent call last):
  File "/home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/jostmva/frontend/blackjack_simulator/routes.py", line 76, in run_simulation_action
    sim.casino_.id = request.form.get('casino_id')
    ^^^^^^^^^^^
AttributeError: 'Simulation' object has no attribute 'casino_'. Did you mean: 'casino'?
2025-10-29 19:36:33,711 INFO: Blackjack Simulator startup [in /home/user/jostmva/frontend/blackjack_simulator/app.py:165]
2025-10-29 19:36:38,149 INFO: Blackjack Simulator startup [in /home/user/jostmva/frontend/blackjack_simulator/app.py:165]
2025-10-29 19:36:53,008 INFO: Running simulation 1 [in /home/user/jostmva/frontend/blackjack_simulator/routes.py:72]
2025-10-29 19:36:53,013 ERROR: Exception on /simulation/1/run_action [POST] [in /home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py:838]
Traceback (most recent call last):
  File "/home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/jostmva/frontend/blackjack_simulator/routes.py", line 76, in run_simulation_action
    sim.casino_.id = request.form.get('casino_id')
    ^^^^^^^^^^^
AttributeError: 'Simulation' object has no attribute 'casino_'. Did you mean: 'casino'?
2025-10-29 19:40:00,340 INFO: Blackjack Simulator startup [in /home/user/jostmva/frontend/blackjack_simulator/app.py:165]
2025-10-29 19:40:05,143 INFO: Blackjack Simulator startup [in /home/user/jostmva/frontend/blackjack_simulator/app.py:165]
2025-10-29 19:40:19,626 INFO: Running simulation 1 [in /home/user/jostmva/frontend/blackjack_simulator/routes.py:72]
2025-10-29 19:40:19,676 ERROR: Exception on /simulation/1/run_action [POST] [in /home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py:838]
Traceback (most recent call last):
  File "/home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/jostmva/.venv/lib/python3.12/site-packages/flask/app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/jostmva/frontend/blackjack_simulator/routes.py", line 99, in run_simulation_action
    "rules": casino.rules
             ^^^^^^^^^^^^
AttributeError: 'Casino' object has no attribute 'rules'
2025-10-29 19:42:02,452 INFO: Blackjack Simulator startup [in /home/user/jostmva/frontend/blackjack_simulator/app.py:165]
2025-10-29 19:42:06,835 INFO: Blackjack Simulator startup [in /home/user/jostmva/frontend/blackjack_simulator/app.py:165]
2025-10-29 19:42:21,261 INFO: Running simulation 1 [in /home/user/jostmva/frontend/blackjack_simulator/routes.py:77]
